x/participationrewards/keeper/rewards_holdings.go

```diff
@@ -94,7 +95,7 @@ func (k Keeper) CalcUserHoldingsAllocations(ctx sdk.Context, zone *icstypes.Zone
 	tokensPerAsset := sdk.NewDecFromInt(zoneAllocation).Quo(sdk.NewDecFromInt(supply.Amount))
 
 	// determine ics rewards to be distributed per token.
-	icsRewardsAddr := sdk.MustAccAddressFromBech32(zone.WithdrawalAddress.Address)
+	icsRewardsAddr, _ := addressutils.AddressFromBech32(zone.WithdrawalAddress.Address, "")
```

Comment:
Correct me if i am wrong, this will still use "quick" as prefix as HRP is empty

Reply:
no, "" means that no check will take place.

Reply:
We don't actually care for the prefix here. 

Although we should probably panic if the bech32 fails to decode, else we'll try to send from nil address and that won't behave well.

Reply:
so just err check and looks good to me

Reply:
I've updated to use the HRP check (as we know it should always be the `zone.AccountPrefix`) and handling the errors appropriately.

We also don't attempt to do this if zone.WithdrawalAddress has not been set yet.

Reply:
amazing works 👍